importer/src/main/java/org/hiero/mirror/importer/migration/AsyncJavaMigration.java

```diff
@@ -63,21 +65,33 @@ with last as (
             where f.installed_rank = last.installed_rank
             """;
 
-    protected final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+    private final ObjectProvider<JdbcTemplate> jdbcTemplateProvider;
+    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
 
     private final AtomicBoolean complete = new AtomicBoolean(false);
     private final String schema;
     private final AtomicBoolean shouldMigrate = new AtomicBoolean(false);
 
     protected AsyncJavaMigration(
             Map<String, MigrationProperties> migrationPropertiesMap,
-            NamedParameterJdbcTemplate namedParameterJdbcTemplate,
+            ObjectProvider<JdbcTemplate> jdbcTemplateProvider,
             String schema) {
         super(migrationPropertiesMap);
-        this.namedParameterJdbcTemplate = namedParameterJdbcTemplate;
+        this.jdbcTemplateProvider = jdbcTemplateProvider;
         this.schema = schema;
     }
 
+    protected final JdbcTemplate getJdbcTemplate() {
+        return jdbcTemplateProvider.getObject();
+    }
+
+    protected final NamedParameterJdbcTemplate getNamedParameterJdbcTemplate() {
```

Comment:
We should only use their Operations interfaces everywhere and not reference the template implementation. We haven't done a good job of that in places, but let's take the time now to do it properly. That is, for fields, methods, etc always use `NamedParameterJdbcOperations` over `NamedParameterJdbcTemplate` and `JdbcOperations` over `JdbcTemplate`.
```suggestion
    protected final JdbcOperations getJdbcOperations() {
        return jdbcOperationsProvider.getObject();
    }

    protected final NamedParameterJdbcOperations getNamedParameterJdbcOperations() {
```